src/ray/core_worker/core_worker.h

```diff
@@ -1077,6 +1083,17 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   /// Whether we are shutting down and not running further tasks.
   bool exiting_ = false;
 
+  /// The `actor_location_resolve_mutex_` field could be mutated concurrently
+  /// due to multi-threading, we need a mutex to protect it.
+  mutable absl::Mutex actor_location_resolve_waiters_mutex_;
```

Comment:
I would suggest putting the variable below under the actor_handles_mutex_ instead of adding a new mutex.

Eventually, these should all be moved out into a separate class for unit testing.

Reply:
I will keep this as it is here and address the request in a separate PR that will come before this.